### PR TITLE
Add anchor tag to URL to signify open page for reloading

### PR DESCRIPTION
**Pull request by MaxLeiter**
example: `https://mylounge.com/#help` opens the help page
`https://mylounge.com/#chan-1` opens the network (because the first connected channel, not the network, is assigned id `0`)

### CODE REVIEW

**Comment by xPaw** [comment]
What's the use case? To get `#chan-1` to work, you'd need to have the channel already there on page load (e.g. logged in). We already re-open active channel on page reloads for logged in users.

**Comment by MaxLeiter** [comment]
Help, settings, and connect windows

**Comment by xPaw** [comment]
But does that solve any real use case?

**Comment by MaxLeiter** [comment]
If I'm on the help page and reload, I expect it to stay on the help page?

**Comment by astorije** [comment]
It will be very useful if we have settings that force a page refresh when changed (like changing the language locale), i.e. when it's much cheaper to just reload than to recompute/repaint the whole app.
It will also be very useful, if not mandatory, when we get to #791. When that happens, loading the app should go to the home page, unless it was a refresh, and it won't make sense to keep track of the last opened tab (this behavior really annoys me at the moment).

In the meantime... maybe it's too soon? :/ I can definitely understand @MaxLeiter's points though.

**Comment by xPaw** [comment]
This will probably actually be needed for push notifications (click on notif -> open channel where it came from).

**Comment by astorije** [comment]
@xPaw, mind taking over this PR as @MaxLeiter will not be able to help for some time?

**Comment by christer88** [comment]
Would this let you specify a channel someone join (for the "login" page in public mode)? e.g. demo.thelounge.org/#testchannel takes the user to #testchannel on the specified network in config file.

**Comment by MaxLeiter** [comment]
@christer88 different feature, but on public mode you can do that like this:

https://demo.thelounge.chat/?join=%23testchannel&nick=example

**Comment by MaxLeiter** [comment]
Bug: if you reload the page on #help, #connect, or #settings the sidebar disappears.

**Comment by astorije** [comment]
@MaxLeiter, @xPaw, what do we want to do with this?

**Comment by MaxLeiter** [comment]
Rebased

**Comment by MaxLeiter** [comment]
there's an issue with history on this -- states seem to be duplicated (so need to hit forward/backward twice)

@thelounge/maintainers - any clues? tried to debug, no success.

**Comment by xPaw** [comment]
@MaxLeiter try doing
```
		.trigger("click", {
			pushState: false,
		});
```

**Comment by MaxLeiter** [comment]
I did - changed nothing.

**Comment by astorije** [comment]
Doesn't `window.location.hash = ...` add an entry in the history, explaining why you have duplicates?

**Comment by MaxLeiter** [comment]
Fixed - PR is ready to go.

**Comment by astorije** [comment]
@xPaw, mind giving a first review to this?

File: client/js/lounge.js [hunk 1]
```diff
@@ -594,7 +594,8 @@ $(function() {
 			}
 		});
 	});
-	if ($("body").hasClass("public")) {
+
+	if ($("body").hasClass("public") && window.location.hash === "#connect") {
```
Comment (xPaw) [inline client/js/lounge.js]: Can we support `#channels` in public mode instead?

**Comment by MaxLeiter** [comment]
Can you elaborate?

File: client/js/lounge.js (hunk 1 above)
Comment (MaxLeiter) [reply client/js/lounge.js]: Can you elaborate?

Comment (MaxLeiter) [reply client/js/lounge.js]: oh, if public mode is trying to connect to channels, populate the channels textbox? Then you're assuming that the user also knows the network, etc. Also, we should probably encourage them to use https://github.com/thelounge/lounge/pull/674

Comment (YaManicKill) [reply client/js/lounge.js]: Yeah, that feels like 2 ways to solve the 1 issue.

**Comment by MaxLeiter** [comment]
@xPaw can you 👍 and we can discuss your request in an issue/PR?

**Comment by MaxLeiter** [comment]
...good to merge?

**Comment by astorije** [comment]
Thanks!

**Pull request merged**